ghost/members-api/lib/repositories/member.js

```diff
@@ -914,6 +914,13 @@ module.exports = class MemberRepository {
             }
         } else if (offerId) {
             offer = await this._offerRepository.getById(offerId, {transacting: options.transacting});
+            if (offer) {
```

Comment:
I don't understand this change - is it directly related to the bug?

Reply:
Its indirectly related, part of the bug wouldn't have happened if we were using the calculated offerId above in the subscription created event instead of using `data.offerId` directly, which just attaches the offer id from metadata to checkout session. Because of that we missed the offer id calculated by stripe coupon as well due to the main bug, otherwise we would have atleast captured coupon offers.
Using calculated `offerId` here ensures that we don't blindly pass in the offer id that we get in data, but only pass in the calculated valid offer id

Reply:
Okay, so do we need this change to fix the bug? If we don't I'd prefer to remove it and open a separate issue for a separate bug so that we can keep this patch as minimal as possible

Reply:
Cool fair, will remove it from here and we can add this in later separately.